Tella/Scenes/Settings/Views/Servers/Uwazi/Step5/UwaziLanguageSelectionView.swift

```diff
@@ -0,0 +1,203 @@
+                            self.presentationMode.wrappedValue.dismiss()
+                        }
+                        LanguageActionButton(type: .ok) {
+                            serverViewModel.handleServerAction()
+                            navigateTo(destination: UwaziSuccessView())
+                        }
+                    }
+                    .padding(.trailing, 20)
+                    .padding(.top, 12)
+                }
+                if serverViewModel.isLoading {
+                    CircularActivityIndicatory()
+                }
+                
+            }
+
+        }
+        .onAppear(perform: {
+            self.serverViewModel.languages.removeAll()
+            self.serverViewModel.getLanguage()
+        })
+        .navigationBarBackButtonHidden(true)
+        .navigationBarItems(leading: backButton)
+    }
+}
+
+struct LanguageActionButton: View {
+    enum ButtonAction {
+        case ok
+        case cancel
+
+        var title: String {
+            switch self {
+            case .ok:
+                return LocalizableSettings.UwaziLanguageOk.localized
+            case .cancel:
+                return LocalizableSettings.UwaziLanguageCancel.localized
+            }
+        }
+        var buttonColor: Color {
+            switch self {
+            case .ok:
+                return Styles.Colors.yellow
+            case .cancel:
+                return Color(UIColor(hexValue: 0xF5F5F5).withAlphaComponent(0.16))
+            }
+        }
+    }
+    let type: ButtonAction
+    var action: () -> Void
+
+    var body: some View {
+        Button(type.title,action: action).buttonStyle(BigButtonStyleForLanguage(color: type.buttonColor))
+    }
+}
+
+struct UwaziLanguageItemView : View {
+
+    var languageItem : UwaziLanguageRow
+    @Binding var selectedLanguage: UwaziLanguageRow?
+    //@StateObject var settingsViewModel :  SettingsViewModel
+
+    @Binding var isPresented : Bool
+
+    @EnvironmentObject private var appViewState: AppViewState
+    @EnvironmentObject private var appModel: MainAppModel
+
+    var body: some View {
+
+        ZStack {
+
+            HStack {
+                VStack(alignment: .leading) {
+                    Text(languageItem.languageName())
+                        .font(.custom(Styles.Fonts.regularFontName, size: 15))
+                        .foregroundColor(.white)
+
+                    Text(languageItem.languageName())
+                        .font(.custom(Styles.Fonts.regularFontName, size: 12))
+                        .foregroundColor(.white)
+                }
+
+                Spacer()
+
+                if isCurrentLanguage(languageItem: languageItem) {
+                    Image("settings.done")
+                }
+
+            }
+            Button("") {
+               // LanguageManager.shared.currentLanguage = languageItem
+                selectedLanguage = languageItem
+                appModel.shouldUpdateLanguage = true
```

Comment:
@rob-horizontal-iOS I don't think that `appModel.shouldUpdateLanguage = true` is necessary here

Reply:
Removed